awsutil/generate_credentials.go

```diff
@@ -100,6 +113,33 @@ func NewCredentialsConfig(opt ...Option) (*CredentialsConfig, error) {
 			}
 		}
 	}
+	c.RoleARN = opts.withRoleArn
+	if c.RoleARN == "" {
```

Comment:
One of the points of using this library is so that we can _not_ use env vars unless explicitly opted in, as they can provide a different identity and level of security than you might want an instance of a plugin to have. That's why only region uses the env var here, and the rest of these should not be used.

Reply:
Scratch that. Apparently these three have always allowed using the env var for some reason. 🤷

Reply:
I do think it would be best to toggle env var reading with the withEnvironmentCredentials option but that would be a potentially breaking change we'd have to synchronize with Vault so for now probably best to leave it alone.

Reply:
What should happen when the user provides static credentials (WithAccessKey, WithSecretKey) and options for generating a Web Identity Role Provider? Should both provider types be added to the credential chain or should an error be return to the user?

Reply:
Both types...the credential chain ordering will prevail, but this allows passing backup if needed

Reply:
Fallthroughs are fickle. Better to just remove the switch.

Reply:
When testing the scenario of having both static credentials and a ec2-instance role provider credential, the static credentials took priority in the credential chain. The reason why I was able to determine this is because each credential provider had different policy access permissions. Is this the expected behavior of the credential chain?